src/handler/get_xpub.c

```diff
@@ -33,7 +33,15 @@ void derive_xpub() {
                                 G_context.bip32_path.length));
     CX_CHECK(init_public_key(&private_key, &public_key));
 
+    const size_t pubk_buflen =
+        sizeof(G_context.pk_info.raw_public_key) / sizeof(G_context.pk_info.raw_public_key[0]);
+    if (pubk_buflen < public_key.W_len) {
+        error = CX_INTERNAL_ERROR;
+        goto end;
+    }
+
     memmove(G_context.pk_info.raw_public_key, public_key.W, public_key.W_len);
+    // pk_info.chain_code is a static buffer with size 32
```

Comment:
Should we check anyway? This might help to prevent bugs in future changes.

Reply:
Done

Reply:
Great! Thanks!